src/Requests/BatchResponseContent.php

```diff
@@ -0,0 +1,142 @@
+use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
+use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;
+
+/**
+ * Class BatchResponseContent
+ *
+ * @package Microsoft\Graph\Core
+ * @copyright 2022 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+class BatchResponseContent implements Parsable
+{
+    /**
+     * @var array
+     */
+    private array $headers = [];
+
+    /**
+     * @var int
+     */
+    private int $statusCode;
+
+    /**
+     * @var array<string, BatchResponseItem>
+     */
+    private array $responses = [];
+
+    /**
+     * @var string
+     */
+    private string $nextLink = "";
+
+    public function __construct() {}
+
+    /**
+     * @return array
+     */
+    public function getHeaders(): array
+    {
+        return $this->headers;
+    }
+
+    /**
+     * @param array $headers
+     */
+    public function setHeaders(array $headers): void
+    {
+        $this->headers = $headers;
+    }
+
+    /**
+     * @return int
+     */
+    public function getStatusCode(): int
+    {
+        return $this->statusCode;
+    }
+
+    /**
+     * @param int $statusCode
+     */
+    public function setStatusCode(int $statusCode): void
+    {
+        $this->statusCode = $statusCode;
+    }
+
+    /**
+     * @return BatchResponseItem[]
+     */
+    public function getResponses(): array
+    {
+        return array_values($this->responses);
+    }
+
+    /**
+     * @param BatchResponseItem[] $responses
+     */
+    public function setResponses(array $responses): void
+    {
+        array_map(fn ($response) => $this->responses[$response->getId()] = $response, $responses);
+    }
+
+    /**
+     * @return string
+     */
+    public function getNextLink(): string
+    {
+        return $this->nextLink;
+    }
+
+    /**
+     * Gets a response for a given request ID
+     * @param string $requestId
+     * @return BatchResponseItem
+     */
+    public function getResponse(string $requestId): BatchResponseItem
+    {
+        if (!array_key_exists($requestId, $this->responses)) {
+            throw new \InvalidArgumentException("No response found for id: {$requestId}");
+        }
+        return $this->responses[$requestId];
+    }
+
+    /**
+     * @param string $nextLink
+     */
+    public function setNextLink(string $nextLink): void
+    {
+        $this->nextLink = $nextLink;
+    }
+
+    public function getFieldDeserializers(): array
+    {
+        return [
+            '@nextLink' => fn (ParseNode $n) => $this->setNextLink($n->getStringValue()),
+            'responses' => fn (ParseNode $n) => $this->setResponses($n->getCollectionOfObjectValues([BatchResponseItem::class, 'create']))
+        ];
+    }
+
+    public function serialize(SerializationWriter $writer): void
+    {
+        $writer->writeStringValue('@nextLink', $this->getNextLink());
```

Comment:
I didn't know that batching supported paging. If it does, shouldn't this be @odata.nextLink instead?

Reply:
Got that from this [Odata spec example](http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_AsynchronousBatchRequests).

Reply:
Also, Graph's docs don't seem to mention paging. Does this mean it's not supported?

Reply:
I've never heard of paging being supported by AGS in the context of batching (the hard limit is 20 requests anyway). @darrelmiller to the rescue here.

Reply:
Removed `@nextLink` for now. In case of confirmation, will return it.